app/controllers/concerns/authentication_and_sso_concerns.rb

```diff
@@ -56,12 +55,13 @@ def validate_session
     @current_user.present?
   end
 
-  def load_user
+  def load_user(skip_terms_check: false)
     if cookies[SignIn::Constants::Auth::ACCESS_TOKEN_COOKIE_NAME]
-      super
+      super(skip_terms_check:)
     else
       @session_object = Session.find(session[:token])
-      @current_user = User.find(@session_object.uuid) if @session_object&.uuid
+      user = User.find(@session_object.uuid) if @session_object&.uuid
+      @current_user = user if skip_terms_check || !user&.needs_accepted_terms_of_use
```

Comment:
I did a mini refactor along the lines of what you suggested, but didn't go all the way. Namely:

- Didn't use the 'session object' function as a memoized function. My reasoning here is that there are various places that refer to this as `@session_object`, and I didn't want to refactor all of the usages at this point. Hopefully I made it more clear that authentication sets this object and sets the user object though
- Didn't use tertiary operation for `current_user`. Not sure why, just didn't think it was important right now.

That being said, I think it's worth looking at this authentication, and how it carefully spins plates with `authenticate` versus `load_user`, and with how the Sign in Service implementation and SSOe are defined basically on top of each other here. Prob a worthy place to refactor for more clarity

Reply:
(btw I also intentionally kept the `cookies` auth completely separate from the other auth, because the `cookies` auth is the Sign in Service one, and the other is SSOe, both completely different authentication methods that we don't want interfering with each other)